src/Storages/Streaming/StorageMaterializedView.cpp

```diff
@@ -537,7 +537,7 @@ void StorageMaterializedView::initBackgroundState()
                                 auto current_failed_sns = getProcessedSNOfStreamingSources(streaming_sources);
                                 recover_sns.clear();
                                 /// If the current SN fails three times in a row, we will skipping some SNs and recover from the next SN to avoid permanent recovery.
-                                if (failed_sn_queue.size() >= 2)
+                                if (failed_sn_queue.size() >= local_context->getSettingsRef().recovery_retry_for_sn_failure)
```

Comment:
`local_context->getSettingsRef().recovery_retry_for_sn_failure - 1`

Reply:
sorry I forget

Reply:
In case `recovery_retry_for_sn_failure == 0`, need special handling

Reply:
thanks, I submit new commit, plz check